src/hooks/useUser.tsx

```diff
@@ -0,0 +1,28 @@
+//hooks
+import { useUserId } from "./UserContext";
+import { useEffect, useState } from "react";
+
+//type
+import { User } from "./type";
+
+const useUser = () => {
```

Comment:
Query hooks naming convention: use<name>Query
Ex: useUserQuery()

Keep the name useCurrentUserQuery as it will always return the info of current user.